hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/AbstractHoodieClient.java

```diff
@@ -124,9 +128,30 @@ public HoodieWriteConfig getConfig() {
     return config;
   }
 
+  private void checkIndexTypeCompatible(String writeIndexType, String persistIndexType) {
+    if (writeIndexType.equals(persistIndexType)) {
```

Comment:
could we strcuture this more easily as a switch statement, so it's easier to extend. or even just a single if -check. any reason the else block below can't be a else if block?

Reply:
can just a single if -check, i will modify it.